### PR TITLE
Unwrap tagged replicated events before publishing to other replicas

### PR DESCRIPTION
**Pull request by johanandren**
Refs #30080

No test coverage because very tricky to combine with the journal proxy needed to run in a multi-jvm test (which requires Java serialization to be enabled)

### CODE REVIEW

**Review by octonato — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala [hunk 1]
```diff
@@ -377,5 +377,10 @@ private[akka] final case class PublishedEventImpl(
     case _                => payload
   }
 
+  override def withoutTags: PublishedEvent = payload match {
+    case Tagged(event, _) => PublishedEventImpl(persistenceId, sequenceNumber, event, timestamp, replicatedMetaData)
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala:381]: copy(payload = event)

**Review by patriknw — APPROVED** [review]
LGTM, with cosmetic suggestion

**Review by chbatey — APPROVED** [review]

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala (hunk 1 above)
Comment (johanandren) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala:381]:
```suggestion
    case Tagged(event, _) => copy(payload = event)
```

**Pull request merged**